apps/condo/domains/common/schema/fields.js

```diff
@@ -131,6 +132,34 @@ const MONEY_AMOUNT_FIELD = {
     },
 }
 
+const POSITIVE_MONEY_AMOUNT_FIELD = {
+    ...MONEY_AMOUNT_FIELD,
+    hooks: {
+        validateInput: ({ resolvedData, addFieldValidationError, fieldPath, listKey }) => {
+            if (resolvedData.hasOwnProperty(fieldPath) && resolvedData[fieldPath] !== null) {
+                const parsedDecimal = Big(resolvedData[fieldPath])
+                if (parsedDecimal.lte(0)) {
+                    addFieldValidationError(`[${listKey.toLowerCase()}:${fieldPath}:negative] Field "${fieldPath}" of "${listKey}" must be greater then 0`)
+                }
+            }
+        },
+    },
+}
+
+const NON_NEGATIVE_MONEY_FIELD = {
+    ...MONEY_AMOUNT_FIELD,
+    hooks: {
+        validateInput: ({ resolvedData, addFieldValidationError, fieldPath, listKey }) => {
+            if (resolvedData.hasOwnProperty(fieldPath) && resolvedData[fieldPath] !== null) {
```

Comment:
`!isNull(get(...))`
and also make sure of handling `undefined` for update operations

Reply:
https://github.com/open-condo-software/condo/pull/807#discussion_r752104789